docs/content/concepts/assets/external-assets.mdx

```diff
@@ -0,0 +1,193 @@
+
+response = requests.request("POST", url, json=payload, headers=headers)
+```
+
+There are also end points for reporting asset observations and asset check evaluations.
+
+### Using Sensors
+
+You can generate events to attach to external assets and provide them to sensors directly, by using the `asset_events` param of <PyObject object="SensorResult"/>.
+
+```python file=/concepts/assets/external_assets/external_asset_using_sensor.py
+import datetime
+
+from dagster import (
+    AssetMaterialization,
+    AssetSpec,
+    Definitions,
+    SensorEvaluationContext,
+    SensorResult,
+    external_asset_from_spec,
+    sensor,
+)
+
+
+def utc_now_str() -> str:
+    return datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%d, %H:%M:%S")
+
+
+@sensor()
+def keep_external_asset_a_up_to_date(context: SensorEvaluationContext) -> SensorResult:
+    # Materialization happened in external system, but is recorded here
+    return SensorResult(
+        asset_events=[
+            AssetMaterialization(
+                asset_key="external_asset_a",
+                metadata={
+                    "source": f'From sensor "{context.sensor_name}" at UTC time "{utc_now_str()}"'
+                },
+            )
+        ]
+    )
+
+
+defs = Definitions(
+    assets=[external_asset_from_spec(AssetSpec("external_asset_a"))],
+    sensors=[keep_external_asset_a_up_to_date],
+)
+```
+
+### Using the Python API
+
+Users can insert events to attach to external assets directly from Dagster's Python API. An example where this would be useful is writing a handrolled Python script to backfill metadata. The relevant API is `report_runless_asset_event` on <PyObject object="DagsterInstance"/>.
+
+```python file=/concepts/assets/external_assets/external_asset_events_using_python_api.py startafter=start_python_api_marker endbefore=end_python_api_marker dedent=4
+from dagster import AssetMaterialization
+
+# instance is a DagsterInstance. Get using DagsterInstance.get()
+instance.report_runless_asset_event(
+    AssetMaterialization(
+        "asset_one", metadata={"nrows": 10, "source": "From this script."}
+    )
+)
+```
+
+### Log events in unrelated ops
+
+A user can log an <PyObject object="AssetMaterialization"/> from a bare op. In this case we use the `log_event` method of <PyObject object="OpExecutionContext"/> to report an asset materialization of an external asset.
+
+```python file=/concepts/assets/external_assets/update_external_asset_via_op.py
+from dagster import (
+    AssetMaterialization,
+    AssetSpec,
+    Definitions,
+    OpExecutionContext,
+    external_asset_from_spec,
+    job,
+    op,
+)
+
+
+@op
+def an_op(context: OpExecutionContext) -> None:
+    context.log_event(AssetMaterialization(asset_key="external_asset"))
+
+
+@job
+def a_job() -> None:
+    an_op()
+
+
+defs = Definitions(
+    assets=[external_asset_from_spec(AssetSpec("external_asset"))], jobs=[a_job]
+)
+```
+
+### What about Source Assets?
```

Comment:
Should this be all the way at the end of the page? I feel like this is info I'd want before I totally dive in.

Reply:
Hmmm ok I went back and forth on this. It is also completely irrelevant to a fresh user.

Reply:
Moved to top. We'll see how that sits.